File: LeetCode/Easy/566 - Reshape the Matrix/566 - Reshape the Matrix.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    const ar = mat.flat();
    if (ar.length !== r * c) return mat;

    const res = [];
    while (ar.length) res.push(ar.splice(0, c));
    
    return res;
};

File: LeetCode/Easy/566 - Reshape the Matrix/566 - Reshape the Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        if (mat.size() * mat[0].size() != r * c) return mat;
        
        vector<vector<int>> vc(r, vector<int>(c));
        int i = 0, j = 0;
        for (auto row : mat) {
            for (auto element : row) {
                vc[i][j] = element;
                j++;
                if (j == c) {
                    i++;
                    j = 0;
                }
            }
        }
        return vc;
    }
};